ci/jenkins/test-vmc.sh

```diff
@@ -185,10 +235,138 @@ function release_static_ip() {
+    # All allow from subnet
+    aws ec2 authorize-security-group-ingress --region $AWS_REGION --group-id $CONTROLPLANE_SGID --protocol "-1" --cidr 0.0.0.0/0
+    # New node security group
+    aws ec2 create-security-group --description capi-node --group-name capi-node --vpc-id $AWS_VPC_ID --region $AWS_REGION
+    export NODE_SGID=$(aws ec2 describe-security-groups --region $AWS_REGION --group-names capi-node --query 'SecurityGroups[0].GroupId' --output text)
+    # All allow from subnet
+    aws ec2 authorize-security-group-ingress --region $AWS_REGION --group-id $NODE_SGID --protocol "-1" --cidr 0.0.0.0/0
+    set -e
+}
+
+function setup_aws_cluster() {
+    env
+    export KUBECONFIG=$KUBECONFIG_PATH
+    if [ -z $K8S_VERSION ]; then
+      export K8S_VERSION=v1.19.1
+    fi
+    export TEST_OS=ubuntu-20.04
+    rm -rf ${GIT_CHECKOUT_DIR}/jenkins || true
+    mkdir -p ~/.aws
+    cat > ~/.aws/config <<EOF
+[default]
+region = $AWS_REGION
+role_arn = $AWS_SERVICE_USER_ROLE_ARN
+source_profile = $AWS_SERVICE_USER_NAME
+output = json
+EOF
+    cat > ~/.aws/credentials <<EOF
+[$AWS_SERVICE_USER_NAME]
+aws_access_key_id = $AWS_ACCESS_KEY_ID
+aws_secret_access_key = $AWS_SECRET_ACCESS_KEY
+EOF
+
+    unset AWS_ACCESS_KEY
+    unset AWS_SECRET_KEY
+    export AWS_CONTROLLER_IAM_ROLE=$AWS_SERVICE_USER_ROLE_ARN
+    export AMI_ID=$(clusterawsadm ami list --kubernetes-version=$K8S_VERSION --os=$TEST_OS --region=$AWS_REGION | grep ami- | awk '{print $5}' | head -1)
+    if [ -z "$AMI_ID" ]; then
+        echo "failed to locate the ami id for kubernetes-version=$K8S_VERSION os=$TEST_OS region=$AWS_REGION" 
+        return 1
+    fi
+
+    echo '=== Copying key pair ==='
+    mkdir -p ${GIT_CHECKOUT_DIR}/jenkins/key
+    cp $AWS_SSH_KEY  "${GIT_CHECKOUT_DIR}/jenkins/key/antrea-ci-key"
+
+    echo "=== setup AWS security group ==="
+    setup_aws_sg
+
+    echo "=== namespace value substitution ==="
+    mkdir -p ${GIT_CHECKOUT_DIR}/jenkins/out
+    cp ${GIT_CHECKOUT_DIR}/ci/cluster-api/aws/templates/* ${GIT_CHECKOUT_DIR}/jenkins/out
+    sed -i "s/VPCID/${AWS_VPC_ID}/g" ${GIT_CHECKOUT_DIR}/jenkins/out/cluster.yaml
+    sed -i "s/SUBNETID/${AWS_SUBNET_ID}/g" ${GIT_CHECKOUT_DIR}/jenkins/out/cluster.yaml
+    sed -i "s/CONTROLPLANESG/${CONTROLPLANE_SGID}/g" ${GIT_CHECKOUT_DIR}/jenkins/out/cluster.yaml
+    sed -i "s/NODESG/${NODE_SGID}/g" ${GIT_CHECKOUT_DIR}/jenkins/out/cluster.yaml
+
+    sed -i "s/CLUSTERNAMESPACE/${CLUSTER}/g" ${GIT_CHECKOUT_DIR}/jenkins/out/cluster.yaml
+    sed -i "s/K8SVERSION/${K8S_VERSION}/g" ${GIT_CHECKOUT_DIR}/jenkins/out/cluster.yaml
+    sed -i "s/AMIID/${AMI_ID}/g" ${GIT_CHECKOUT_DIR}/jenkins/out/cluster.yaml
+    sed -i "s/CLUSTERNAME/${CLUSTER}/g" ${GIT_CHECKOUT_DIR}/jenkins/out/cluster.yaml
+    sed -i "s/SSHAUTHORIZEDKEYS/default/g" ${GIT_CHECKOUT_DIR}/jenkins/out/cluster.yaml
+    sed -i "s/CLUSTERNAMESPACE/${CLUSTER}/g" ${GIT_CHECKOUT_DIR}/jenkins/out/namespace.yaml
+
+    sleep 15s
```

Comment:
Why do you need 15s here? If the step 'setup_aws_sg' takes some time, you may move it to line 298.

Reply:
we need to add a wait of 15s for the management cluster to be initialised, we cannot move further until this is done.

Reply:
A bit confused, too. When did it trigger or start the initialization of the management cluster? I thought the management cluster was created in advance, and it should be ready before creating any workload clusters.